CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+# include some defines automatically made by qpm
+include(qpm_defines.cmake)
+
+cmake_minimum_required(VERSION 3.22)
+project(${COMPILE_ID})
+
+# c++ standard
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED 20)
+
+# define that stores the actual source directory
+set(SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/src)
+set(INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
+
+# compile options used
+add_compile_options(-frtti -fPIE -fPIC -fexceptions)
+add_compile_options(-Wall -Wextra -Werror -Wno-unused-function)
+# compile definitions used
+add_compile_definitions(VERSION=\"${MOD_VERSION}\")
```

Comment:
Not sure why you pass in MOD_VERSION and MOD_ID differently if you could just pass them in as CLI terms... This seems like a large waste and are functionally different.

Reply:
I'm sorry, what do you mean exactly here? these variables are passed from qpm_defines.cmake and are dependent on the qpm.json, so that is why they are passed as compile definitions here

Reply:
I suppose that will do, just a bit weird and I'd argue is also a misnomer-- `MOD_ID` instead of just `ID` or better yet, `PACKAGE_ID` etc. Just a small peeve, especially since we had previously used `MOD_ID` before swapping over to `ID` so I wanted to make sure nothing was doing any old and legacy stuff.